.github/workflows/build.yml

```diff
@@ -232,14 +232,44 @@ jobs:
           # We already ran ESLint in a separate step
           DISABLE_AUTOMATIC_ESLINT=true
           yarn build
+      - run: tar -czf build.tar.gz build
+      - uses: actions/upload-artifact@v3
+        with:
+          name: e2e-build
+          path: build.tar.gz
+
+  e2e-test:
+    needs: [e2e-build]
+    name: e2e-test-shard-${{ matrix.shard }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        shard: [1, 2, 3, 4]
```

Comment:
my only concern is whether we're going to run up against some github action limits, otherwise this looks like a pretty straightforward win to me 👍

Reply:
simc is currently doing 113 actions on every PR, so this should be okay. I think it gets a bit more worrying depending on how long these may run.

Reply:
👍 we'll find out, worst case we turn the e2e stuff off if it causes problems